### PR TITLE
Add some french translation

### PR DESCRIPTION
**Pull request by pfongkye**

### CODE REVIEW

File: README-fr.md [hunk 1]
```diff
@@ -280,69 +280,69 @@ Consultez les liens suivants pour avoir une meilleure idée de ce qui est attend
 * [L'entretien de conception de système](http://www.hiredintech.com/system-design)
 * [Introduction aux entretiens sur l'architecture et la conception de systèmes](https://www.youtube.com/watch?v=ZgdS0EUmn70)
 
-## System design interview questions with solutions
+## Questions d'entretien sur la conception de système avec solutions
```
Comment (spuyet) [inline README-fr.md]: Not sure that is the right translation, I'd say instead:
```suggestion
## Questions et solutions sur la conception de système
```
WDYT ?

File: README-fr.md [hunk 2]
```diff
@@ -280,69 +280,69 @@ Consultez les liens suivants pour avoir une meilleure idée de ce qui est attend
 * [L'entretien de conception de système](http://www.hiredintech.com/system-design)
 * [Introduction aux entretiens sur l'architecture et la conception de systèmes](https://www.youtube.com/watch?v=ZgdS0EUmn70)
 
-## System design interview questions with solutions
+## Questions d'entretien sur la conception de système avec solutions
 
-> Common system design interview questions with sample discussions, code, and diagrams.
+> Questions d'entretien courantes sur la conception de système avec des exemples de discussions, de code et de diagrammes.
```
Comment (spuyet) [inline README-fr.md]: Same for this one, I suggest:
```suggestion
> Questions fréquentes sur la conception de système avec des exemples de discussions, de code et de diagrammes.
```
WDYT ?

File: README-fr.md [hunk 3]
```diff
@@ -280,69 +280,69 @@ Consultez les liens suivants pour avoir une meilleure idée de ce qui est attend
...
-| Design a key-value store for a search engine | [Solution](solutions/system_design/query_cache/README.md) |
-| Design Amazon's sales ranking by category feature | [Solution](solutions/system_design/sales_rank/README.md) |
-| Design a system that scales to millions of users on AWS | [Solution](solutions/system_design/scaling_aws/README.md) |
-| Add a system design question | [Contribute](#contributing) |
+| Concevoir Pastebin.com (ou Bit.ly) | [Solution](solutions/system_design/pastebin/README.md) |
+| Concevoir la *timeline* et la recherche de Twitter (ou le fil d'activité et la recherche de Facebook) | [Solution](solutions/system_design/twitter/README.md) |
+| Concevoir un *crawler Web* | [Solution](solutions/system_design/web_crawler/README.md) |
+| Concevoir Mint.com | [Solution](solutions/system_design/mint/README.md) |
+| Concevoir la structure de données d'un réseau social | [Solution](solutions/system_design/social_graph/README.md) |
+| Concevoir un magasin de clés-valeurs d'un moteur de recherche | [Solution](solutions/system_design/query_cache/README.md) |
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
| Concevoir un tableau associatif pour un moteur de recherche | [Solution](solutions/system_design/query_cache/README.md) |

```

File: README-fr.md [hunk 4]
```diff
@@ -280,69 +280,69 @@ Consultez les liens suivants pour avoir une meilleure idée de ce qui est attend
...
+| Concevoir le classement des ventes par catégorie de Amazon | [Solution](solutions/system_design/sales_rank/README.md) |
+| Concevoir un système qui peut monter à des millions d'utilisateurs sur AWS | [Solution](solutions/system_design/scaling_aws/README.md) |
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
| Concevoir un système qui peut supporter des millions d'utilisateurs sur AWS | [Solution](solutions/system_design/scaling_aws/README.md) |

```

File: README-fr.md [hunk 5]
```diff
@@ -280,69 +280,69 @@ Consultez les liens suivants pour avoir une meilleure idée de ce qui est attend
...
-### Design the data structures for a social network
+### Concevoir la structure de données d'un réseau social
 
-[View exercise and solution](solutions/system_design/social_graph/README.md)
+[Voir l'exercice et la solution](solutions/system_design/social_graph/README.md)
 
 ![Imgur](http://i.imgur.com/cdCv5g7.png)
 
-### Design a key-value store for a search engine
+### Concevoir un magasin de clés-valeurs d'un moteur de recherche
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
### Concevoir un tableau associatif pour un moteur de recherche

```

File: README-fr.md [hunk 6]
```diff
@@ -280,69 +280,69 @@ Consultez les liens suivants pour avoir une meilleure idée de ce qui est attend
...
-### Design Amazon's sales ranking by category feature
+### Concevoir le classement des ventes par catégorie d'Amazon
 
-[View exercise and solution](solutions/system_design/sales_rank/README.md)
+[Voir l'exercice et la solution](solutions/system_design/sales_rank/README.md)
 
 ![Imgur](http://i.imgur.com/MzExP06.png)
 
-### Design a system that scales to millions of users on AWS
+### Concevoir un système qui peut monter à des millions d'utilisateurs sur AWS
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
### Concevoir un système qui peut supporter des millions d'utilisateurs sur AWS

```

**Review by spuyet — CHANGES_REQUESTED** [review]
Sorry for the delay. First of all, thanks for your kind contribution. I've made some comments, I let you have a look :wink:

**Comment by pfongkye** [comment]
Hi @spuyet I've taken into account your comments. Thanks :+1:

File: README-fr.md [hunk 7]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
 
-* Who is going to use it?
-* How are they going to use it?
-* How many users are there?
-* What does the system do?
-* What are the inputs and outputs of the system?
-* How much data do we expect to handle?
-* How many requests per second do we expect?
-* What is the expected read to write ratio?
+* Qui vont l'utiliser?
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
* Qui va l'utiliser?

```

File: README-fr.md [hunk 8]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
+* Comment vont-ils l'utiliser?
+* Combien y aura-t-il d'utilisateurs?
+* Que fait le système?
+* Quelles sont les entrées-sorties du système?
+* Quelle est la quantité de données qu'on s'attend à traiter?
+* Combien de requêtes par seconde qu'on s'attend à avoir?
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
* Combien de requêtes par seconde attendues ?

```

File: README-fr.md [hunk 9]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
+* Quel est le ratio lecture/écriture qu'on s'attend à avoir?
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
* Quel est le ratio lecture/écriture qu'on attendu ?

```

File: README-fr.md [hunk 10]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
+    * [MD5](solutions/system_design/pastebin/README.md) et [Base62](solutions/system_design/pastebin/README.md)
+    * Collisions de hachage
+    * SQL ou NoSQL
+    * Schéma de base de données
+* Traduire l'URL de hachage en URL complète
+    * Recherche en base de données
+* API et conception orientée objet
 
-### Step 4: Scale the design
+### Etape 4: Mettre à l'echelle la conception
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
### Etape 4: Mise à l'echelle la conception

```

File: README-fr.md [hunk 11]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
 
-Identify and address bottlenecks, given the constraints.  For example, do you need the following to address scalability issues?
+Etant donné les contraintes, identifiez et éliminez les goulets d'étranglement. Par exemple pour régler des problèmes de *scalabilité*, avez-vous besoin de?
```
Comment (spuyet) [inline README-fr.md:264]:
```suggestion
Etant donné les contraintes, identifiez et éliminez les goulots d'étranglement. Par exemple pour régler des problèmes de *scalabilité*, avez-vous besoin de ?

```

File: README-fr.md [hunk 12]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
-* Horizontal scaling
-* Caching
-* Database sharding
+* Répartiteur de charge
+* Mise à l'echelle horizontale
+* Mise en cache
+* Eclatement de base de données (*Database sharding*)
 
-Discuss potential solutions and trade-offs.  Everything is a trade-off.  Address bottlenecks using [principles of scalable system design](#index-of-system-design-topics).
+Discutez des solutions potentielles et des compromis. Toute chose est un compromis. Réglez les goulets d'étranglement en utilisant [les principes de conception de systèmes à grande échelle](#index-of-system-design-topics).
```
Comment (spuyet) [inline README-fr.md:271]:
```suggestion
Discutez des solutions potentielles et des compromis. Toute chose est un compromis. Réglez les goulots d'étranglement en utilisant [les principes de conception de systèmes à grande échelle](#index-of-system-design-topics).

```

File: README-fr.md [hunk 13]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
+## Comment approcher une question d'entretien sur la conception de système
 
-> How to tackle a system design interview question.
+> Comment aborder une question d'entretien sur la conception de système.
 
-The system design interview is an **open-ended conversation**.  You are expected to lead it.
+L'entretien sur la conception de système est une **conversation ouverte**. On attend de vous que vous la meniez.
 
-You can use the following steps to guide the discussion.  To help solidify this process, work through the [System design interview questions with solutions](#system-design-interview-questions-with-solutions) section using the following steps.
+Vous pouvez utiliser les étapes suivantes pour mener la discussion. Pour vous habituer au processus et vous y imprégner, pratiquez les [Questions d'entretien sur la conception de système avec solutions](#system-design-interview-questions-with-solutions) comme suit.
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
Vous pouvez utiliser les étapes suivantes pour mener la discussion. Pour vous habituer au processus et vous en imprégner, pratiquez les [Questions d'entretien sur la conception de système avec solutions](#system-design-interview-questions-with-solutions) comme suit.

```

File: README-fr.md [hunk 14]
```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
+On s'attend à ce que les candidats plus expérimentés aient une plus grande connaissance sur la conception de système, d'autant plus s'ils ont le rôle d'architecte ou team lead. Il est fort probable que les plus grandes entreprises technologiques fassent passer un ou plusieurs tours d'entretiens sur la conception de système.
 
-Start broad and go deeper in a few areas.  It helps to know a little about various key system design topics.  Adjust the following guide based on your timeline, experience, what positions you are interviewing for, and which companies you are interviewing with.
+Commencez par avoir une approche générale qui couvre un large spectre avant d'approfondir sur quelques domaines. Ça aide d'avoir un peu de connaissance sur les différents sujets clés de conception de système. Vous pouvez ajuster ce guide à votre préparation en se basant sur votre chronologie d'entretien, votre expérience, le(s) poste(s) que vous visez et dans quelles entreprises.
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
Commencez par avoir une approche générale qui couvre un large spectre avant d'approfondir sur certains domaines. Ça aide d'avoir un peu de connaissance sur les différents sujets clés de conception de système. Vous pouvez ajuster ce guide à votre préparation en se basant sur votre chronologie d'entretien, votre expérience, le(s) poste(s) que vous visez et dans quelles entreprises.

```

**Review by spuyet — CHANGES_REQUESTED** [review]
I've made a last review and added some minor comments. Could you have a look @pfongkye ?

File: README-fr.md (hunk 11 above)
Comment (pfongkye) [reply README-fr.md:264]: Both are correct. I think we can say `goulet` or `goulot`

Comment (spuyet) [reply README-fr.md:264]: I just checked and you're right, let's keep it like this :wink:

File: README-fr.md [hunk 15]
```diff
@@ -154,195 +154,205 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
...
+    * [MD5](solutions/system_design/pastebin/README.md) et [Base62](solutions/system_design/pastebin/README.md)
+    * Collisions de hachage
+    * SQL ou NoSQL
+    * Schéma de base de données
+* Traduire l'URL de hachage en URL complète
+    * Recherche en base de données
+* API et conception orientée objet
 
-### Back-of-the-envelope calculations
+### Etape 4: Mise à l'echelle la conception
```
Comment (spuyet) [inline README-fr.md]:
```suggestion
### Etape 4: Mise à l'echelle de la conception
```

**Comment by spuyet** [comment]
Thanks for this major contribution @pfongkye, really appreciated :+1:

**Pull request merged**